src/main/java/seedu/address/logic/parser/ParserUtil.java

```diff
@@ -131,7 +133,9 @@ public static Allergy parseAllergy(String allergy) throws ParseException {
     public static Date parseDate(String date) throws ParseException {
         requireNonNull(date);
         String trimmedDate = date.trim();
-
+        if (trimmedDate.toLowerCase().equals("none")) {
```

Comment:
It's good that the code allows for different variations and is not case sensitive. Another function that can be used to do this is equalsIgnoreCase()